sky/clouds/service_catalog/images/provisioners/skypilot.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# Popular third-party dependencies
+sudo apt-get install jq -y
+
+# Function to find processes using a specific file
+mylsof() {
+  p=$(for pid in /proc/{0..9}*; do
+    i=$(basename "$pid")
+    for file in "$pid"/fd/*; do
+      link=$(readlink -e "$file")
+      if [ "$link" = "$1" ]; then
+        echo "$i"
+      fi
+    done
+  done)
+  echo "$p"
+}
+
+# Stop and disable unattended-upgrades
+sudo systemctl stop unattended-upgrades || true
+sudo systemctl disable unattended-upgrades || true
+sudo sed -i 's/Unattended-Upgrade "1"/Unattended-Upgrade "0"/g' /etc/apt/apt.conf.d/20auto-upgrades || true
+
+# Release the dpkg lock if held
+p=$(mylsof "/var/lib/dpkg/lock-frontend")
+echo "$p"
+sudo kill -9 $(echo "$p" | tail -n 1) || true
+sudo rm /var/lib/dpkg/lock-frontend
+sudo pkill -9 dpkg
+sudo pkill -9 apt-get
+
+# Configure dpkg
+sudo dpkg --configure --force-overwrite -a
+
+# Create necessary directories
+mkdir -p ~/sky_workdir
+mkdir -p ~/.sky/
+mkdir -p ~/.sky/sky_app
+
+# Configure SSH
+mkdir -p ~/.ssh
+touch ~/.ssh/config
+
+# Install Miniconda
+curl -o Miniconda3-Linux-x86_64.sh https://repo.anaconda.com/miniconda/Miniconda3-py310_23.11.0-2-Linux-x86_64.sh
+bash Miniconda3-Linux-x86_64.sh -b
+eval "$(~/miniconda3/bin/conda shell.bash hook)"
+conda init
+conda config --set auto_activate_base true
+conda activate base
+
+# Ensure conda is initialized in ~/.bashrc
+grep "# >>> conda initialize >>>" ~/.bashrc || { conda init && source ~/.bashrc; }
+
+# Create conda environment with Python 3.10
+echo "Creating conda env with Python 3.10"
+conda create -y -n skypilot-runtime python=3.10
+conda activate skypilot-runtime
+
+# Set up Python environment
+$([ -s ~/.sky/python_path ] && cat ~/.sky/python_path 2> /dev/null || which python3) -m venv ~/skypilot-runtime --system-site-packages
```

Comment:
We probably should get rid of this `--system-site-packages` for our hosted image to make it more robust to user changing packages in the base python env, as 
```suggestion
$([ -s ~/.sky/python_path ] && cat ~/.sky/python_path 2> /dev/null || which python3) -m venv ~/skypilot-runtime
```
See details in: https://github.com/skypilot-org/skypilot/issues/4097